r/R/dplyr-functions.R

```diff
@@ -694,6 +694,14 @@ nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption
 }
 
 nse_funcs$log <- nse_funcs$logb <- function(x, base = exp(1)) {
+  if (inherits(base, "Expression")) {
+    return(Expression$create("logb_checked", x, base))
+  }
+
+  if (!is.numeric(base) || length(base) != 1) {
+    arrow_not_supported("base with length != 1")
+  }
+
```

Comment:
To your other question, you could check here whether `x` is numeric length 1 and return `base::log(x, base)`. I'd add a comment showing the use case, just to be clear.